### ISSUE TITLE
Could you provide the model and test image that you used?

### ISSUE THREAD

**Issue by justfouw** [cfellicious/loss-visualization#1]
Could you provide the model and test image that you used? May be the download link is also OK.

**Comment by justfouw** [comment]
Thank you very much. I notice that the model files have been uploaded in `model` directories. All python scripts can be run. But it is time consuming because the loss should be calculated each literation. So could you post the result image (e.g. loss landscape) that you have .

**Comment by cfellicious** [comment]
Yes, it is very time consuming. I have uploaded the csv files and the sample images. There are more csv files that the ones that currently output from the code. I will upload the Octave code for visualization later. The loss is visualized using the log loss values.

**Comment by justfouw** [comment]
@cfellicious It is so glad of you. I am of great interest for this paper (visualizing loss surface). I have read this paper times. However, I am a little confused as well. Even after read your code. So I think could us create one  Communication channel, e,g. email to discuss  detained information. You can find my email in my profile page (gmail).

**Comment by cfellicious** [comment]
It would be better to post your doubts here as it could help someone else too.

**Comment by justfouw** [comment]
@cfellicious Yes.
To my understanding, if we want to plot 2D loss surface, we need : 
 - use net weight of any iteration as basic weight, e.g initial weight, final weight, or any one between them.
 - 'sample' new weights based on two random directional vectors and basic weight. E.g. if we plot loss surface with size of 50x50. We need sample 2,500 times. 
- use all sampled weights to calculate loss and then plot surface

Q: In your code, you sample random vectors from Gaussian distribution, v1, and v2. Then you create 'vector_grid1' by uniformly partition [v1, -v1] into 'Step' slices. why ?

**Comment by cfellicious** [comment]
To create a grid, I need to create the intermediate points. And for that,
1. I consider the grid from -v1,-v2 to v1,v2. The origin will represent the results of the actual model. i.e v1 and v2 will be 0
2. I then partition the line from -v1 to v1 into the required number of points and similarly for v2. This would create the grid.

Example consider v1 as 3 and v2 as 3. I create the points for v1. -3,-2,-1,0,1,2,3 and for v2 as -3,-2,-1,0,1,2,3. Then I create a grid based on these points. So the grid would be from -3,-3 to 3,3. All the points on the grid would be created based on the step slices that I created.

I hope this answers your question.

**Comment by cfellicious** [comment]
To create a grid, I need to create the intermediate points. And for that,
1. I consider the grid from -v1,-v2 to v1,v2. The origin will represent the results of the actual model. i.e v1 and v2 will be 0
2. I then partition the line from -v1 to v1 into the required number of points and similarly for v2. This would create the grid.

Example consider v1 as 3 and v2 as 3. I create the points for v1. -3,-2,-1,0,1,2,3 and for v2 as -3,-2,-1,0,1,2,3. Then I create a grid based on these points. So the grid would be from -3,-3 to 3,3. All the points on the grid would be created based on the step slices that I created.

I hope this answers your question.

**Comment by justfouw** [comment]
@ Yes, I agree with you. I mean that this kind of configuration, -v1~v1, -v2~v2, is different from this paper. In fact, in the paper did not mention 'mesh' or grid, at all.  For your setting, what your plot in fact  is centered by $\theta^*$ with one specified size, 50x50, e.g.

**Comment by cfellicious** [comment]
Yes, the paper has not mentioned anything  such as a mesh or grid.
The paper mentions that there are two gaussian vectors that are used to create the landscape. If you check Fig. 1 in the paper you can see a 3-D visualization. The logarithmic loss is computed on the z axis but the x and y axes are formed by v1 and v2(gaussian vectors). We need to modify the weights of v1 and v2 inorder to obtain the loss at each point. 
I have created the 51x51 grid. You can create the grid based on any odd numbered size you want. It could be 101x101 or even 1001x1001. I initially did a 25x25 grid and the resolution was poor so I changed it to 51x51.

**Comment by justfouw** [comment]
@cfellicious Yes, I agree. Some detailed information of this visualization method is missed. I have contracted with the author. This work is of great use value. I will keep on it.

**Comment by cfellicious** [comment]
@justfouw Do you have any updates on the project? Did the authors give you any input?

**Comment by justfouw** [comment]
@cfellicious The original code has been available now.

**Comment by cfellicious** [comment]
> @cfellicious The original code has been available now.

Could you post the link to the original code please

**Comment by justfouw** [comment]
@cfellicious https://github.com/tomgoldstein/loss-landscape
